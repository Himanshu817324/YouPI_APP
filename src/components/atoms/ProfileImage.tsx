import React from 'react';
import { View, Text, Image, TouchableOpacity, StyleSheet } from 'react-native';

interface ProfileImageProps {
  imageUrl?: string;
  fullName?: string;
  size?: number;
  onPress?: () => void;
  showEditIcon?: boolean;
}

const ProfileImage: React.FC<ProfileImageProps> = ({
  imageUrl,
  fullName = '',
  size = 80,
  onPress,
  showEditIcon = false,
}) => {
  const getInitials = (name: string) => {
    if (!name) return 'U';
    
    const names = name.trim().split(' ');
    if (names.length === 1) {
      return names[0].charAt(0).toUpperCase();
    }
    
    return (names[0].charAt(0) + names[names.length - 1].charAt(0)).toUpperCase();
  };

  const containerStyle = {
    width: size,
    height: size,
    borderRadius: size / 2,
  };

  const textStyle = {
    fontSize: size * 0.4,
    fontWeight: 'bold' as const,
  };

  const renderImage = () => {
    if (imageUrl) {
      return (
        <Image
          source={{ uri: imageUrl }}
          style={[styles.image, containerStyle]}
          resizeMode="cover"
        />
      );
    }

    return (
      <View style={[styles.placeholder, containerStyle, { backgroundColor: '#3ED3A3' }]}>
        <Text style={[styles.placeholderText, textStyle, { color: 'white' }]}>
          {getInitials(fullName)}
        </Text>
      </View>
    );
  };

  if (onPress) {
    return (
      <TouchableOpacity onPress={onPress} style={styles.container}>
        {renderImage()}
        {showEditIcon && (
          <View style={[styles.editIcon, { bottom: -2, right: -2 }]}>
            <Text style={styles.editIconText}>✏️</Text>
          </View>
        )}
      </TouchableOpacity>
    );
  }

  return (
    <View style={styles.container}>
      {renderImage()}
      {showEditIcon && (
        <View style={[styles.editIcon, { bottom: -2, right: -2 }]}>
          <Text style={styles.editIconText}>✏️</Text>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'relative',
  },
  image: {
    borderWidth: 2,
    borderColor: '#3ED3A3',
  },
  placeholder: {
    borderWidth: 2,
    borderColor: '#3ED3A3',
    justifyContent: 'center',
    alignItems: 'center',
  },
  placeholderText: {
    fontWeight: 'bold',
  },
  editIcon: {
    position: 'absolute',
    backgroundColor: '#3ED3A3',
    borderRadius: 12,
    width: 24,
    height: 24,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 2,
    borderColor: 'white',
  },
  editIconText: {
    fontSize: 12,
  },
});

export default ProfileImage;
