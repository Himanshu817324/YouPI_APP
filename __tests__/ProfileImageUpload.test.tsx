/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import ProfileImage from '../src/components/atoms/ProfileImage';

describe('ProfileImage Component', () => {
  test('renders with initials when no image provided', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage fullName="John Doe" size={80} />
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders with image when imageUrl provided', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage 
        imageUrl="https://example.com/profile.jpg" 
        fullName="John Doe" 
        size={80} 
      />
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders with edit icon when showEditIcon is true', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage 
        fullName="John Doe" 
        size={80} 
        showEditIcon={true}
        onPress={() => {}}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  test('generates correct initials for single name', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage fullName="John" size={80} />
    );
    expect(tree).toMatchSnapshot();
  });

  test('generates correct initials for multiple names', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage fullName="John Michael Doe" size={80} />
    );
    expect(tree).toMatchSnapshot();
  });

  test('handles empty name gracefully', () => {
    const tree = ReactTestRenderer.create(
      <ProfileImage fullName="" size={80} />
    );
    expect(tree).toMatchSnapshot();
  });
});
