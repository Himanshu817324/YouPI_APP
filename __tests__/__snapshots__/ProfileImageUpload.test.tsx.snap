// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage Component generates correct initials for multiple names 1`] = `null`;

exports[`ProfileImage Component generates correct initials for single name 1`] = `null`;

exports[`ProfileImage Component handles empty name gracefully 1`] = `null`;

exports[`ProfileImage Component renders with edit icon when showEditIcon is true 1`] = `null`;

exports[`ProfileImage Component renders with image when imageUrl provided 1`] = `null`;

exports[`ProfileImage Component renders with initials when no image provided 1`] = `null`;
